cmake/SetupCompilers.cmake

```diff
@@ -102,12 +102,18 @@ if ( MSVC )
 endif()
 
 if (RAJA_ENABLE_CUDA)
+  # When we require cmake 3.8+, replace this with setting CUDA_STANDARD
+  if(CUDA_VERSION_MAJOR GREATER "8")
+    set(RAJA_NVCC_STD "c++14")
+  else()
+    set(RAJA_NVCC_STD "c++11")
+  endif()
   if(CMAKE_BUILD_TYPE MATCHES Release)
-    set(RAJA_NVCC_FLAGS -O2; -restrict; -arch ${RAJA_CUDA_ARCH}; -std c++11; --expt-extended-lambda; -ccbin; ${CMAKE_CXX_COMPILER} CACHE LIST "")
+    set(RAJA_NVCC_FLAGS -O2; -restrict; -arch ${RAJA_CUDA_ARCH}; -std ${RAJA_NVCC_STD}; --expt-extended-lambda; -ccbin; ${CMAKE_CXX_COMPILER} CACHE LIST "")
   elseif(CMAKE_BUILD_TYPE MATCHES RelWithDebInfo)
-    set(RAJA_NVCC_FLAGS -g; -G; -O2; -restrict; -arch ${RAJA_CUDA_ARCH}; -std c++11; --expt-extended-lambda; -ccbin ${CMAKE_CXX_COMPILER} CACHE LIST "")
-  elseif(CMAKE_BUILD_TYPE MATCHES Debug)
-    set(RAJA_NVCC_FLAGS -g; -G; -O0; -restrict; -arch ${RAJA_CUDA_ARCH}; -std c++11; --expt-extended-lambda; -ccbin ${CMAKE_CXX_COMPILER} CACHE LIST "")
+    set(RAJA_NVCC_FLAGS -g; -G; -O2; -restrict; -arch ${RAJA_CUDA_ARCH}; -std  ${RAJA_NVCC_STD}; --expt-extended-lambda; -ccbin ${CMAKE_CXX_COMPILER} CACHE LIST "")
+  else(CMAKE_BUILD_TYPE MATCHES Debug)
```

Comment:
I don't think our "else" case should be adding `-g`, `-G`, and `-O0`. It should probably just add `-std`, `-ccbin` and `--expt-extended-lambda`.

Reply:
Fair point.  There are only four possible values here, so I've added the fourth and changed the else branch to RelWithDebInfo, which is the usual default for package managers like spack anyway.